server/datastore/mysql/software.go

```diff
@@ -2872,243 +2278,753 @@ last_vpp_install AS (
-				}
-			}
+	for _, s := range softwareUninstalls {
+		bySoftwareID[s.SoftwareID] = s
+	}
+
+	vppInstallsStmt := fmt.Sprintf(`
+        WITH upcoming_vpp_install AS (
+            SELECT
+                ua.execution_id AS last_install_install_uuid,
+                -- ua.host_id,
+                ua.created_at AS last_install_installed_at,
+                vaua.adam_id AS vpp_app_adam_id,
+                -- vaua.platform,
+                'pending_install' AS status
+            FROM
+                upcoming_activities ua
+            INNER JOIN
+                vpp_app_upcoming_activities vaua ON ua.id = vaua.upcoming_activity_id
+            LEFT JOIN (
+                upcoming_activities ua2
+                INNER JOIN vpp_app_upcoming_activities vaua2 ON ua2.id = vaua2.upcoming_activity_id
+            ) ON ua.host_id = ua2.host_id AND
+                vaua.adam_id = vaua2.adam_id AND
+                vaua.platform = vaua2.platform AND
+                ua.activity_type = ua2.activity_type AND
+                (ua2.priority < ua.priority OR ua2.created_at > ua.created_at)
+            WHERE
+                ua.host_id = :host_id AND
+                ua.activity_type = 'vpp_app_install' AND
+                ua2.id IS NULL
+        ),
+        last_vpp_install AS (
+            SELECT
+                hvsi.command_uuid AS last_install_install_uuid,
+                -- hvsi.host_id,
+                hvsi.created_at AS last_install_installed_at,
+                hvsi.adam_id AS vpp_app_adam_id,
+                -- hvsi.platform,
+                %s
+            FROM
+                host_vpp_software_installs hvsi
+            LEFT JOIN
+                nano_command_results ncr ON ncr.command_uuid = hvsi.command_uuid
+            LEFT JOIN
+                host_vpp_software_installs hvsi2 ON hvsi.host_id = hvsi2.host_id AND
+                    hvsi.adam_id = hvsi2.adam_id AND
+                    hvsi.platform = hvsi2.platform AND
+                    hvsi2.removed = 0 AND
+                    (hvsi.created_at < hvsi2.created_at OR (hvsi.created_at = hvsi2.created_at AND hvsi.id < hvsi2.id))
+            WHERE
+                hvsi.host_id = :host_id AND
+                hvsi.removed = 0 AND
+                hvsi2.id IS NULL AND
+                NOT EXISTS (
+                    SELECT 1
+                    FROM
+                        upcoming_activities ua
+                    INNER JOIN
+                        vpp_app_upcoming_activities vaua ON ua.id = vaua.upcoming_activity_id
+                    WHERE
+                        ua.host_id = hvsi.host_id AND
+                        vaua.adam_id = hvsi.adam_id AND
+                        vaua.platform = hvsi.platform AND
+                        ua.activity_type = 'vpp_app_install'
+                )
+        )
+        SELECT
+			*
+		FROM
+            (SELECT * FROM upcoming_vpp_install UNION SELECT * FROM last_vpp_install) AS lvia
```

Comment:
I don't believe we need to add vulnerabilities joins for this. After VPP software is installed on the host we track those through host software.